OpenRA.Game/GlobalChat.cs

```diff
@@ -0,0 +1,342 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Meebey.SmartIrc4net;
+using OpenRA;
+using OpenRA.Primitives;
+
+namespace OpenRA.Chat
+{
+	public enum ChatConnectionStatus { Disconnected, Connecting, Connected, Disconnecting, Joined, Error }
+	public enum ChatMessageType { Message, Notification }
+
+	public sealed class ChatUser
+	{
+		public readonly string Name;
+		public bool IsOp;
+		public bool IsVoiced;
+
+		public ChatUser(string name, bool isOp, bool isVoice)
+		{
+			Name = name;
+			IsOp = isOp;
+			IsVoiced = isVoice;
+		}
+	}
+
+	public sealed class ChatUserComparer : IComparer<string>
```

Comment:
Use one on the static members on `System.StringComparer` and save yourself this class.